File: app/src/main/java/com/example/vladi/consultagit/io/ApiService.java
package com.example.vladi.consultagit.io;

import com.example.vladi.consultagit.io.response.PullsResponse;
import com.example.vladi.consultagit.io.response.RepositoriesResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import retrofit2.http.Url;

public interface ApiService {

    @GET("search/repositories")
    Call<RepositoriesResponse> getRepositories(
            @Query("q") String language,
            @Query("sort") String sort,
            @Query("page") String page);

    @GET("repos/{owner}/{repo}/pulls")
    Call<PullsResponse> getPulls(
            @Path ("owner") String authorName,
            @Path ("repo") String repositoryName);

    }

File: app/src/main/java/com/concrete/challenge/domain/io/APIService.kt
package com.concrete.challenge.domain.io

import com.concrete.challenge.data.PullRequestEntity
import com.concrete.challenge.data.UserEntity
import com.concrete.challenge.domain.io.response.RepositoriesResponse
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface APIService {

    @GET("search/repositories?q=language:Java&sort=stars&page=1")
    suspend fun getRepositories(
        @Query("page") page: Int
    ): RepositoriesResponse

    @GET("users")
    suspend fun getUser(
        @Query("username") username: String
    ): List<UserEntity>

    @GET("repos/{owner}/{repo}/pulls")
    suspend fun getPullRequests(
        @Path("owner") username: String,
        @Path("repo") repositoryName: String
    ): List<PullRequestEntity>

}